go.mod

```diff
@@ -1,16 +1,18 @@
 module github.com/celestiaorg/celestia-node
 
-go 1.23.0
+go 1.23.1
```

Comment:
Better to bump to 1.23.2 to include Go fix for https://github.com/celestiaorg/celestia-node/pull/3790

Also, maybe drop `godebug` below? 🤔 That's the question to the node team, not you Rootul.

Reply:
`go mod tidy` auto bumped to Go 1.23.1 but I agree Go 1.23.2 is probably a better target for ya'll considering it fixes https://github.com/golang/go/issues/69333